arangod/Replication2/ReplicatedLog/LogFollower.cpp

```diff
@@ -352,7 +357,7 @@ auto replicated_log::LogFollower::GuardedFollowerData::checkCommitIndex(
         << newLITK << ".";
     _lowestIndexToKeep = newLITK;
     // TODO do we want to call checkCompaction here?
-    std::ignore = checkCompaction();
+    // std::ignore = checkCompaction();
```

Comment:
I don't think the answer is no. One reason I did this change for now is that the tests otherwise cannot catch the bug this PR fixes, because this hides it. The other one being that I think we might not want to do it synchronously here; but I haven't thought it through yet.